src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -614,7 +614,7 @@ private void ProcessEvents()
                                 // parent and previousEventName is the name of the directory to be removed.
                                 foreach (WatchedDirectory child in previousEventParent.Children)
                                 {
-                                    if (child.Name == previousEventName)
+                                    if (previousEventName.SequenceEqual(child.Name))
```

Comment:
nit, it would be clearer to use `Equals(child.Name, StringComparison.Ordinal)` I think